parser/src/cfg/mod.rs

```diff
@@ -1051,13 +1062,50 @@ fn parse_layer_indexes(exprs: &[SpannedLayerExprs], expected_len: usize) -> Resu
         let layer_expr = subexprs.next().ok_or_else(|| {
             anyhow_span!(expr, "deflayer requires a name and {expected_len} item(s)")
         })?;
-        let layer_name = match expr_type {
-            SpannedLayerExprs::DefsrcMapping(_) => layer_expr
-                .atom(None)
-                .ok_or_else(|| {
-                    anyhow_expr!(layer_expr, "layer name after {DEFLAYER} must be a string")
-                })?
-                .to_owned(),
+        let (layer_name, icon) = match expr_type {
+            SpannedLayerExprs::DefsrcMapping(_) => match layer_expr {
+                SExpr::Atom(name_span) => (name_span.t.to_owned(), None),
+                SExpr::List(name_opts_span) => {
+                    let listt = &name_opts_span.t;
+                    let mut list = listt.iter();
+                    let name = list.next().ok_or_else(|| {anyhow_span!(name_opts_span,"deflayer requires a string name within this pair of parenthesis (or a string name without any)")})?
+                        .atom(None).ok_or_else(|| {anyhow_expr!(layer_expr, "layer name after {DEFLAYER} must be a string when enclosed within one pair of parentheses")})?;
+                    let mut icon = "";
+                    //todo: add hashmap for future options, currently only parse icons
+                    let mut layer_opts: HashMap<String, String> = HashMap::default();
```

Comment:
Please put this code in a new file, e.g. `parser/src/cfg/layer_opts.rs`. And try to see if we can deduplicate the code.

Reply:
moved